File: packages/lib-shared-py/src/lib_shared/core/utils/pick/_pick.py
from typing import Any, Mapping, Unpack, cast

from lib_shared.core.utils.pick._pick_models import _PickModel, _PickParamsModel
from pydash import pick


def _pick[T: Mapping[str, Any]](*params: Unpack[_PickParamsModel[T]]) -> _PickModel[T]:
    value, keys = params
    return cast(T, pick(value, keys))

File: packages/lib-model-js/src/core/utils/pick/pick.ts
import { _pick } from '@lib/shared/core/utils/pick/_pick';
import { type PickModel, type PickParamsModel } from '@lib/shared/core/utils/pick/pick.models';

export const pick = <TType extends object, TKey extends string>(
  ...params: PickParamsModel<TType, TKey>
): PickModel<TType, TKey> => _pick<TType, TKey>(...params);
